File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.py
from collections import deque

def solution(cacheSize, cities):
    cache = deque([])

    answer = 0
    for city in cities:
        new_city = city.upper()
        if (new_city not in cache):
            answer += 5
            if (len(cache) == cacheSize):
                cache.append(new_city)
                cache.popleft()
            else:
                cache.append(new_city)
        else:
            cache.remove(new_city)
            cache.append(new_city)
            answer += 1
            
    return answer

File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.java
import java.util.Deque;
import java.util.LinkedList;

class Solution {
    public int solution(int cacheSize, String[] cities) {
        int answer = 0;
        Deque<String> cache = new LinkedList<>();
        
        for (String city : cities) {
            String tmp = city.toUpperCase();
            
            if (cache.contains(tmp)) {
                cache.remove(tmp);
                cache.add(tmp);
                answer += 1;
            } else {
                if (cache.size() == cacheSize) {
                    cache.add(tmp);
                    cache.removeFirst();
                    answer += 5;
                } else {
                    cache.add(tmp);
                    answer += 5;
                }
            }
            
        }

        return answer;
    }
}